### PR TITLE
testing/gradle: new aport

### PR DESCRIPTION
**Pull request by asiekierka**
A Java-based build system

### CODE REVIEW

**Comment by skrzyp** [comment]
well...

File: testing/gradle/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+license="ASL-2.0"
+depends="openjdk8-jre-base"
+makedepends=""
+subpackages="$pkgname-doc"
+source="http://services.gradle.org/distributions/$pkgname-$pkgver-all.zip
+        gradle"
+builddir="$srcdir/$pkgname-$pkgver"
+pkghome="/usr/share/java/$pkgname"
+
+prepare() {
```
Comment (jirutka) [inline testing/gradle/APKBUILD]: When you define custom `prepare` function, please always invoke `default_prepare || return 1`, usually as the first command in the function.

File: testing/gradle/gradle [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+APP_NAME="Gradle"
+APP_BASE_NAME=`basename "$0"`
```
Comment (jirutka) [inline testing/gradle/gradle]: Backticks syntax for subshell is deprecated for decades, please use `$(…)`.

File: testing/gradle/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+url="http://gradle.org/"
+arch="noarch"
+license="ASL-2.0"
+depends="openjdk8-jre-base"
+makedepends=""
+subpackages="$pkgname-doc"
+source="http://services.gradle.org/distributions/$pkgname-$pkgver-all.zip
+        gradle"
+builddir="$srcdir/$pkgname-$pkgver"
+pkghome="/usr/share/java/$pkgname"
```
Comment (jirutka) [inline testing/gradle/APKBUILD]: Custom global variables should be prefixed with underscore (i.e. `_pkghome`).

File: testing/gradle/gradle (hunk 2 above)
Comment (asiekierka) [inline testing/gradle/gradle]: Sorry for that one - this was derived from Gradle's own /usr/bin script, just trimmed down heavily as we don't need things like Darwin compatibility.

**Comment by jirutka** [comment]
The build fails because of bad checksum:

```
>>> gradle: Checking sha512sums...
gradle-3.3-all.zip: OK
gradle: FAILED
sha512sum: WARNING: 1 of 1 computed checksums did NOT match
```

**Comment by clandmeter** [comment]
Closed due to inactivity.

**Pull request closed without merging**